src/components/Query.tsx

```diff
@@ -28,6 +29,10 @@ class QueryHandler extends Component<QueryHandlerProps, QueryHandlerState> {
   executeQuery = (opts?: Partial<OperationContext>) => {
     this.unsubscribe();
 
+    if (this.props.skip) {
```

Comment:
Per your suggestions @kitten moved this after the call to `unsubscribe`. Moreover, this would return before `fetching` ever gets set to `true`, so I don't think we need to handle resetting `fetching` to `false`.